### PR TITLE
Added update! functionality

### PR DESCRIPTION
**Pull request by cbecker**
Hi Tim, this adds an `update!()` method which I found useful for certain situations, such as the one I added to the README.md file.

### CODE REVIEW

File: src/ProgressMeter.jl [hunk 1]
```diff
@@ -70,11 +70,28 @@ function next!(p::Progress)
     end
 end
 
+function next!(p::Progress)
```
Comment (timholy) [inline src/ProgressMeter.jl]: Now calling just `next!` will never update the progress bar, meaning that it will break existing code.

Comment (cbecker) [inline src/ProgressMeter.jl]: oh sorry you are right, I forgot to fix this.

Comment (cbecker) [inline src/ProgressMeter.jl]: Fixed now, see below.

**Comment by timholy** [comment]
Ah, there's also a conflict with current master. You were probably behind when you implemented this---if the package maintainer (me) doesn't tag after each commit, then you can be modifying an outdated package. (You can protect yourself against that by checking out `master` first.)

Sorry about that. I just tagged a new release---it was definitely overdue. I made it a minor release, to reflect all the good work folks have been putting into this.

Anyway, you might need to do a `git fetch; git rebase -i origin/master`. More info is [here](http://docs.julialang.org/en/latest/manual/packages/#squashing-and-rebasing).

**Pull request closed without merging**